dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/FileStorageResource.java

```diff
@@ -0,0 +1,392 @@
+        this.contentletMetadataAPI = contentletMetadataAPI;
+    }
+
+    public FileStorageResource() {
+        this(APILocator.getFileStorageAPI(), new WebResource(),
+                new MultiPartUtils(), APILocator.getContentletAPI(), APILocator.getContentletMetadataAPI());
+    }
+
+    @PUT
+    @Path("/metadata/basic")
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    public final Response generateMetadata(@Context final HttpServletRequest request,
+                                      @Context final HttpServletResponse response,
+                                               final FormDataMultiPart multipart) throws IOException, JSONException {
+
+        final Tuple2<Map<String,Object>, List<File>> bodyAndBinaries =
+                this.multiPartUtils.getBodyMapAndBinariesFromMultipart(multipart);
+        final List<File> files            = bodyAndBinaries._2();
+        final Map<String, Object> bodyMap = bodyAndBinaries._1();
+
+        final ImmutableMap.Builder<String, Object> bodyResultBuilder = new ImmutableMap.Builder<>();
+        if (!UtilMethods.isSet(files)) {
+
+            throw new BadRequestException("Must send files to generate the metadata");
+        }
+
+        for (final File file : files) {
+
+            if (bodyMap.isEmpty()) {
+
+                bodyResultBuilder.put(file.getName(), this.fileStorageAPI.generateRawBasicMetaData(file));
+            } else {
+
+                generateMetadata(bodyMap, bodyResultBuilder, file, false);
+            }
+        }
+
+        return Response.ok(new ResponseEntityView(bodyResultBuilder.build())).build();
+    }
+
+    private GenerateMetaDataConfiguration buildConfig (final Map<String, Object> bodyMap,
+                                                        final boolean full) throws JSONException {
+
+        final GenerateMetaDataConfiguration.Builder builder = new GenerateMetaDataConfiguration.Builder();
+        builder.full(full);
+
+        if (bodyMap.containsKey("cacheKey")) {
+
+            builder.cache(() -> (String) bodyMap.get("cacheKey"));
+        }
+
+        if (bodyMap.containsKey("override")) {
+
+            builder.override(ConversionUtils.toBoolean(bodyMap.get("override").toString(), false));
+        }
+
+        if (bodyMap.containsKey("maxLength")) {
+
+            builder.maxLength(ConversionUtils.toLong(bodyMap.get("maxLength"), FileStorageAPI.configuredMaxLength()));
+        }
+
+        if (bodyMap.containsKey("fields")) {
+
+            final JSONArray fieldJSONArray = (JSONArray)bodyMap.get("fields");
+            final Set<String> fieldSet = new HashSet<>();
+            for (int i = 0; i < fieldJSONArray.length(); ++i) {
+
+                fieldSet.add(fieldJSONArray.getString(i));
+            }
+            builder.metaDataKeyFilter(fieldSet::contains);
+        }
+
+        if (bodyMap.containsKey("file")) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "file" appears 4 times in this file; the first occurrence is on line 158](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)